jcl/src/java.base/share/classes/java/lang/ClassLoader.java

```diff
@@ -2101,13 +2105,27 @@ static void loadLibrary(Class<?> caller, String libName) {
 	}
 }
 
-static long findNative(ClassLoader loader, String entryName) {
+/*[IF JAVA_SPEC_VERSION >= 24]*/
+static long findNative1(ClassLoader loader, String entryName, Class<?> cls, String javaName) {
+	long address = findNative0(loader, entryName);
```

Comment:
Does the spec specify this order of checks? As in, if we look for a method that doesnt exist in a class that doesnt have native access, which error shows up?

Reply:
The spec does not explicitly define the order of checks, but we can infer it from the available OpenJDK tests. Currently, we are aligned with the RI since we pass the majority of failing tests in #20354.